### ISSUE TITLE
ruby upgrade issues? (ie, missing app/assets/config/manifest.js file, puma can't load application)

### ISSUE THREAD

**Issue by visualist** [coyote-team/coyote#436]
In an effort to build and run Coyote on Ubuntu, I am trying this with newer versions of Ruby and the various gems that this project depends on. I've also updated Dockerfile in order to pull-in newer versions (as-is, the Docker file and older versions of everything did not work, or at least I could not find the matchings set of older ruby/gem/os versions that worked for me under Ruby 2.4.1). I will attach to this ticket the updated Dockerfile and Gemfile that I used to create.

OS: Ubuntu 18.04 (AWS EC2 generic AMI) deployed to t2.medium (4g memory)
Ruby: 2.5.1
Updated Gemfile/Gemfile.lock and Dockerfile accordingly to fit this environment

"docker-compose up" results in the following error:
`Unable to load application: Sprockets::Railtie::ManifestNeededError: Expected to find a manifest file in app/assets/config/manifest.js`

This was the first stumbling block, and when I created this file a certain way, I was able to achieve a working system.

Bottom line: I think these issues are more 'upgrade framework'  and 'upgrade gems' issues than anything else. So documenting this experience and the pitfalls along the way.

**Comment by visualist** [comment]
Dockerfile
```
FROM ruby:2.5.1
MAINTAINER Mike Subelsky <mike@subelsky.com>
RUN apt-get update -qq && apt-get install -y build-essential libpq-dev nodejs software-properties
-common

# in upgraded configuration: jessie-pgdg replaces xenial-pgdg
RUN add-apt-repository "deb http://apt.postgresql.org/pub/repos/apt/ jessie-pgdg main"
RUN wget --quiet -O - https://www.postgresql.org/media/keys/ACCC4CF8.asc | apt-key add -
RUN apt-get update
RUN apt-get install -y postgresql-client-9.6
RUN apt-get install -y coreutils
RUN apt-get install -y default-libmysqlclient-dev
#                      (instead of libmysqlclient-dev)

ADD . /coyote
WORKDIR /coyote

RUN gem install bundler -v 1.16.6

RUN bundler -v; gem list | grep bundler
RUN bundle config

# pre-install a few of the gems (sometimes runs into memory problems while building on small syst
ems)
RUN gem install nokogiri -v '1.10.9' --source 'https://rubygems.org/'
RUN gem install unf_ext -v 0.0.7.6
RUN gem install sassc -v 2.2.1
RUN gem install bcrypt -v 3.1.13
RUN bundle install
```

**Comment by visualist** [comment]
Gemfile
```
# frozen_string_literal: true
  
source 'https://rubygems.org'
ruby '2.5.1'

git_source(:github) do |repo_name|
  repo_name = "#{repo_name}/#{repo_name}" unless repo_name.include?("/")
  "https://github.com/#{repo_name}.git"
end

# TEMPORARILY INSTALLING THIS TO FACILITATE MIGRATING FROM MYSQL
# TODO: remove this dependency after migration is complete
gem 'mysql2', require: false

gem 'rails', '5.2.0'

gem 'analytical' # analytics
gem 'apipie-rails', '0.5.10'
gem 'audited'
gem 'autoprefixer-rails'
gem 'aws-sdk-s3', require: false
gem 'bootsnap'
gem 'devise' # user auth
gem 'dotenv-rails'
gem 'easymarklet', git: 'https://github.com/seeread/easymarklet.git', ref: '53829a6'
gem 'great_pretender' # TODO: Move back to dev after redesign
gem 'haml-rails', '>= 1.0.0'
gem 'iconv'
gem 'jquery-rails', '>= 4.0.4'
gem 'jsonapi-rails'

gem 'nio4r'
gem 'rails-dom-testing', '~> 2.0'
gem 'nokogiri'

gem 'kaminari'
gem 'language_list'
gem 'lightbox2-rails', github: 'johansmitsnl/lightbox2-rails' # for resource lightboxes on index pages
gem 'markdown-rails'
gem 'metamagic' # meta
...
```
(no changes after the nokogiri line)

Diff looks as follows:
```
$ git diff Gemfile
diff --git a/Gemfile b/Gemfile
index d6c161d..7e4c506 100644
--- a/Gemfile
+++ b/Gemfile
@@ -1,7 +1,7 @@
 # frozen_string_literal: true
 
 source 'https://rubygems.org'
-ruby '2.4.1'
+ruby '2.5.1'
 
 git_source(:github) do |repo_name|
   repo_name = "#{repo_name}/#{repo_name}" unless repo_name.include?("/")
@@ -15,7 +15,7 @@ gem 'mysql2', require: false
 gem 'rails', '5.2.0'
 
 gem 'analytical' # analytics
-gem 'apipie-rails'
+gem 'apipie-rails', '0.5.10'
 gem 'audited'
 gem 'autoprefixer-rails'
 gem 'aws-sdk-s3', require: false
@@ -28,6 +28,11 @@ gem 'haml-rails', '>= 1.0.0'
 gem 'iconv'
 gem 'jquery-rails', '>= 4.0.4'
 gem 'jsonapi-rails'
+
+gem 'nio4r'
+gem 'rails-dom-testing', '~> 2.0'
+gem 'nokogiri'
+
 gem 'kaminari'
 gem 'language_list'
 gem 'lightbox2-rails', github: 'johansmitsnl/lightbox2-rails' # for resource lightboxes on index pages
```

**Comment by visualist** [comment]
Also, I prebuilt a Gemfile.lock using the same version of ruby on-system as requested in Dockerfile.

**Comment by visualist** [comment]
In the spirit of [this SO answer](https://stackoverflow.com/questions/58339607/why-does-rails-fails-to-boot-with-expected-to-find-a-manifest-file-in-app-asse), I created a manifest.js file, but with some differences in order get Coyote to build and run.

manifest.js
```
// JS and CSS bundles
//
//= link_directory ../javascripts .js
//= link_directory ../stylesheets .css


// Images and fonts so that views can link to them
//
//  link_tree ../fonts
//  link_tree ../images

//= link favicon.ico
```

**Comment by visualist** [comment]
My remaining problem was with the favicon stuff, which I couldn't figure out how to solve between gems, and references like the above. So I punted and, just for now, removed the references to favicon in the view-code for these two files (ie, one line diffs, removed "= render 'favicon'" from each file):
- app/views/layouts/application.html.slim
- app/views/layouts/coyote_producer.haml

**Comment by sinabahram** [comment]
Just a note here that when Docker gets upgraded for this project, this will hopefully go away, as we should just have a Ruby/Rails docker, Mysql/postgress docker, and possibly an nginx one to sit in front.

**Comment by visualist** [comment]
Having gotten past these hurdles, I can build and run the application as documented:
- docker-compose build
- docker-compose up

**Comment by visualist** [comment]
@sinabahram - agree that do a real upgrade, properly, will likely look different from my hacks, but thought it could be useful, here, to document what I've done to get it running. Thanks!

**Comment by sinabahram** [comment]
, totally useful. I was just noting for future issue review so we can close once that’s done.

**Comment by flipsasser** [comment]
The asset issues were resolved w/the upgrade to Rails 6 and the fixed Docker config. Thanks @visualist for the notes and I'm going to close this out - please open a new issue if you run into additional problems!
